HeroesPowerPlant/SetIdTableEditor/SetIdTableFunctions.cs

```diff
@@ -31,112 +31,96 @@ public static StageEntry[] ReadStageListData(string FileName)
-                TableEntry TemporaryEntry;
-                if (isShadow)
-                    TemporaryEntry = ReadShadowTableEntry(tableReader, objectEntries);
-                else
-                    TemporaryEntry = ReadHeroesTableEntry(tableReader, objectEntries);
-
-                tableEntries.Add(TemporaryEntry);
-            }
-
-            tableReader.Close();
+                while (reader.BaseStream.Position != reader.BaseStream.Length)
+                    tableEntries.Add(ReadHeroesTableEntry(reader, objectEntries));
 
             return tableEntries;
         }
 
-        private static TableEntry ReadHeroesTableEntry(BinaryReader tableReader, Dictionary<(byte, byte), ObjectEntry> objectEntries)
+        private static TableEntry ReadHeroesTableEntry(EndianBinaryReader reader, Dictionary<(byte, byte), ObjectEntry> objectEntries)
         {
-            TableEntry temporaryEntry = new TableEntry
+            var entry = new TableEntry
             {
-                values0 = Switch(tableReader.ReadUInt32()),
-                values1 = Switch(tableReader.ReadUInt32()),
-                values2 = Switch(tableReader.ReadUInt32())
+                values0 = reader.ReadUInt32(),
+                values1 = reader.ReadUInt32(),
+                values2 = reader.ReadUInt32()
             };
-            tableReader.BaseStream.Position += 6;
+            reader.BaseStream.Position += 6;
 
-            byte objList = tableReader.ReadByte();
-            byte objType = tableReader.ReadByte();
+            byte objList = reader.ReadByte();
+            byte objType = reader.ReadByte();
 
             if (objectEntries.ContainsKey((objList, objType)))
-                temporaryEntry.objectEntry = objectEntries[(objList, objType)];
+                entry.objectEntry = objectEntries[(objList, objType)];
             else
-                temporaryEntry.objectEntry = new ObjectEntry()
+                entry.objectEntry = new ObjectEntry()
                 {
                     Type = objType,
                     List = objList,
                     Name = "Unknown/Unused"
                 };
 
-            return temporaryEntry;
+            return entry;
         }
 
-        private static TableEntry ReadShadowTableEntry(BinaryReader tableReader, Dictionary<(byte, byte), ObjectEntry> objectEntries)
+        private static TableEntry ReadShadowTableEntry(EndianBinaryReader tableReader, Dictionary<(byte, byte), ObjectEntry> objectEntries)
         {
-            TableEntry temporaryEntry = new TableEntry();
+            var entry = new TableEntry();
 
             byte objType = tableReader.ReadByte();
             byte objList = tableReader.ReadByte();
             tableReader.ReadInt16();
-            temporaryEntry.values0 = Switch(tableReader.ReadUInt32());
-            temporaryEntry.values1 = Switch(tableReader.ReadUInt32());
+            entry.values0 = tableReader.ReadUInt32();
+            entry.values1 = tableReader.ReadUInt32();

```

Comment:
This was the source of breakage. Endian was supposed to be swapped. Also in Write. Accounted for in https://github.com/igorseabra4/HeroesPowerPlant/commit/911d53379d037081bd75bb396f9fc0d1b759dd32